packages/internal/src/priority-senders.js

```diff
@@ -0,0 +1,85 @@
+import { E, Far } from '@endo/far';
+
+const { Fail, quote: q } = assert;
+
+export const PRIORITY_SENDERS_NAMESPACE_RE = /^[a-zA-Z0-9_-]{1,50}$/;
+
+/**
+ *
+ * @param {ERef<import('./lib-chainStorage.js').StorageNode>} sendersNode
+ */
+export const makePrioritySendersManager = sendersNode => {
+  /**
+   * address to tuple with storage node and set of namespaces that requested priority
+   *
+   * @type {Map<string, readonly [node: StorageNode, namespaces: Set<string>]>}
+   */
+  const addressRecords = new Map();
+
+  const refreshVstorage = (node, namespaces) => {
+    E(node).setValue(
+      // if the list set is empty, the string will be '' and thus deleted from IAVL
+      [...namespaces.keys()].sort().join(','),
+    );
+  };
+
+  return Far('prioritySenders manager', {
+    /**
+     * @param {string} namespace
+     * @param {string} address
+     * @returns {Promise<void>}
+     */
+    add: async (namespace, address) => {
+      if (!namespace.match(PRIORITY_SENDERS_NAMESPACE_RE)) {
+        throw Fail`invalid namespace ${q(namespace)}`;
+      }
+      // get or make record
+      const record = await (async () => {
+        const extant = addressRecords.get(address);
+        if (extant) {
+          return extant;
+        }
+        const node = await E(sendersNode).makeChildNode(address);
```

Comment:
for good measure in case the `sendersNode` was not created with the right options, should we force `{ sequence: false }` in the options here too? For what it's worth, it's the nodes where it matters.

Reply:
worthwhile for documentation at least. I'll make the change